src/ducks/apps/index.js

```diff
@@ -268,10 +268,12 @@ export function fetchIconsProgressively() {
   return async (dispatch, getState) => {
     const apps = getState().apps.list
     const iconsMap = new Map() // slug: icon
-    apps.forEach(async (app, index) => {
+    let updateCounter = 0
+    apps.forEach(async app => {
       const iconData = await _getIcon(app.iconToLoad)
       iconsMap.set(app.slug, iconData)
-      if (iconsMap.size % 10 === 0 || index === apps.length - 1) {
+      updateCounter++
+      if (iconsMap.size % 10 === 0 || updateCounter === apps.length) {
```

Comment:
I don't understand the change :/

Reply:
As my loop function is asynchronous here, we can't relate on the index argument. So we have to  handle a dedicated counter to know if we are at the end of the update or not.